src/frontend/dash/circlePacker.ts

```diff
@@ -1,203 +1,351 @@
+                    circle.destroy();
+                    return false;
                 }
             }
+            return true;
         }
-    }
+    );
+
+    return loggedIn.filter(
+        entry => !placedCircles.find(
+            circle => circle instanceof MemberCircle && circle.name == entry[0]
+        )
+    );
 }
 
-function isVacant(circle, x, y) {
-    for(let circle1 of placedCircles) {
-        if(getDistanceFrom(circle1, x, y) + 0.0001 < circle.r + circle1.r)
-            return false;
+export function placeCircles(circles : Circle[]) {
+    circles = circles.sort((circleA, circleB) => circleB.r - circleA.r);
+
+    const bounds = getBounds();
+
+    const newPlacedCircles = [];
+    let maxNewBoundSpace = Math.max(...circles.map(circle => circle.r));
+
+    for(const circle of circles) {
+        const offsetX = Math.random() * maxNewBoundSpace + circle.r;
+        const offsetY = Math.random() * maxNewBoundSpace + circle.r;
+        
+        circle.position.x = Math.random() > 0.5 ? offsetX + bounds.maxX : bounds.minX - offsetX;
+        circle.position.y = Math.random() > 0.5 ? offsetY + bounds.maxY : bounds.minY - offsetY;
+
+        newPlacedCircles.push(circle);
+        maxNewBoundSpace += circle.r;
     }
-    return true;
+
+    placedCircles.push(...newPlacedCircles);
 }
-// const sizeMin = 1;
-// const sizeMax = 30;
-
-// function clampSize(value) {
-//     return Math.max(sizeMin, Math.min(sizeMax, value));
-// }
-
-// assumes unplacedCircles is EMPTY
-// and placedCircles is FILLED
-export function placeCircles(circles: Circle[]) {
-    minX = 0;
-    minY = 0;
-    maxX = 0;
-    maxY = 0;
-    targetMaxX = 0;
-    targetMaxY = 0;
-    targetMinX = 0;
-    targetMinY = 0;
-    // normalize
-    placedCircles = [];
-    const unplacedCircles = circles.sort((a, b) => b.r - a.r);
-    // circles.splice(Math.floor(circles.length/2),0,circles.splice(circles.findIndex(circle=>circle instanceof ClockCircle),1)[0])
-
-    let sizeSum = 0;
-
-    for(let circle = unplacedCircles.shift(); circle; circle = unplacedCircles.shift()) {
-        // circle.x = circle.y = 0;
-        // circle.r = clampSize(circle.r);
-        circle.r -= (circle.r - (sizeSum += circle.r)/(placedCircles.length+1)) * deltaAvg;
-        placeCircle(circle);
-        placedCircles[placedCircles.length] = circle;
-        let targetMaxX1 = maxX = Math.max(maxX, circle.x + circle.r);
-        let targetMaxY1 = maxY = Math.max(maxY, circle.y + circle.r);
-        let targetMinX1 = minX = Math.min(minX, circle.x - circle.r);
-        let targetMinY1 = minY = Math.min(minY, circle.y - circle.r);
-
-        targetMaxX = Math.max(targetMaxX1, targetMaxY1*aspectRatio);
-        targetMaxY = Math.max(targetMaxY1, targetMaxX1/aspectRatio);
-        targetMinX = Math.min(targetMinX1, targetMinY1*aspectRatio);
-        targetMinY = Math.min(targetMinY1, targetMinX1/aspectRatio);
+
+export function updateCircles(time : number) {
+    if(time > 100) return;
```

Comment:
did you mean `<`? Also shouldn't this be handled by whatever is calling your function, not the function itself?

Reply:
The function is there because setInterval pauses when the tab is not visible, which would cause the circle to travel a certain velocity for a large amount of time. The only way I could prevent that was by making the function detect when the time is too large then skip itself.